dask/utils.py

```diff
@@ -843,6 +843,8 @@ def infer_storage_options(urlpath, inherit_storage_options=None):
     if protocol == 'file':
         # Special case parsing file protocol URL on Windows according to:
         # https://msdn.microsoft.com/en-us/library/jj710207.aspx
+        # urlsplit does not work with ?
+        path = urlpath if "?" in urlpath else path
```

Comment:
@mrocklin , as things stand, the ? becomes a query indicator indeed: for a file:, the path beyond ? is cut off, for s3:, you get `unexpected keyword argument 'url_query'`.
@roman-kh , we don't have a http: file open (although we thought about it), principally because there is no way to run glob.